docs/content/usage/concepts.md

```diff
@@ -0,0 +1,113 @@
+- As per Azure [template limits](https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/azure-subscription-service-limits#template-limits), it is not possible to perform more than 800 deployments using a single template. This limitation is more likely to be encountered in a template orchestrated approach, in case of a more complex solution, leveraging many AVM modules. Note: this limitation doesn't apply to Terraform.
+- Some application and data layer configurations cannot be done through template orchestration without workarounds, e.g., uploading files, creating Entra ID (AAD) objects, etc.. Note: some of these limitations don't apply to Terraform.
+- The deploying context (user or service principal) must have permissions of all resource deployments that are defined in the solution template.
+
+### Pipeline orchestration
+
+Pipeline orchestration allows you to "solution workflow" that uses platform specific pipeline capabilities (e.g., GitHub workflows, Azure DevOps pipelines) to provision individual modules, where each job deploys one module. By defining dependencies in between jobs you can make sure your resources are deployed in order. Parallelization is achieved by using a pool of pipeline agents (runners) that run the jobs, while accounting for all dependencies defined.
+
+This approach is more complex and is typically used for larger deployments. Pipeline orchestration allows you to define your entire deployment process in a single pipeline, with more granularity and control (approvals and RBAC).
+
+![PipelineOrchestration]({{% siteparam base %}}/images/usage/concepts/pipelineOrchestration.png?height=400px "Pipeline Orchestration")
+
+#### Advantages of pipeline orchestration
+
+- The deployment of an individual resource is very simple
+- Most CI/CD systems provide you with a visual representation of the deployment flow
+- If a deployment fails, you can re-run it independently
+- Splitting your solution workflow into individual jobs makes the solution easier to troubleshoot
+- Different deployment jobs can use different principals, resulting in a more granular permission model
+
+#### Disadvantages of pipeline orchestration
+
+- Each deployment needs its own job, and therefore, its own agent. As a consequence, parallel resource deployments require multiple agents.
+- Passing information from one deployment to another requires passing information from one agent to another.
+- As each agent job has to start up and check out the code first, it generally runs slower than native, template-orchestrated deployments.
+
+Note: pipeline orchestrated deployments are not to be confused with the inner workings of CI/CD pipelines.
```

Comment:
Hey @matebarabas - I think these sections are very Bicep specific. They don't really work for Terraform. Perhaps we could have separate sub-pages for each language to avoid confusion. Happy to work with you on this 😄

Reply:
Yep, good observation, fully agree. We concluded on the same, hence, we're hiding this for now and will bring it back when we get to this stage and we can properly describe what's applicable for what. We're only focusing on the "solution development" articles in this release.